### PR TITLE
CHUI-10: Add handler for place order button and complete the purchase

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

Adds the `onClick` handler to the "Place Order" button to start the purchase pipeline.

This PR depends on vtex-apps/checkout-payment#9

#### How should this be manually tested?

[Workspace](https://placeorder--checkoutio.myvtex.com/cart/add?sku=289).

You should be able to place the order after correctly filling all the checkout steps on the workspace above.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

**Review by marcoskwkm — APPROVED** [review]

**Comment by kevinch** [comment]
@lucasecdb i got stuck at this step:
Screen Shot 2020-06-04 at 15 19 45

Refresh did the trick. Do not know what happened but wanted to register here.

File: react/PlaceOrder.tsx [hunk 1]
```diff
@@ -1,14 +1,138 @@
...
+  const { orderForm } = useOrderForm()
+  const { value, referenceValue, interestValue } = useOrderPayment()
+  const {
+    culture: { currency },
+    rootPath = '',
+  } = useRuntime()
+
+  const [placingOrder, setPlacingOrder] = useState(false)
+  const [cardFormIframe] = useState(
+    () => document.getElementById('chk-card-form')! as HTMLIFrameElement
```
Comment (kevinch) [inline react/PlaceOrder.tsx:22]: What is this `HTMLIFrameElement`?

File: react/PlaceOrder.tsx [hunk 2]
```diff
@@ -1,14 +1,138 @@
...
-import styles from './PlaceOrder.css'
+const { useOrderForm } = OrderForm
+const { useOrderPayment } = OrderPayment
 
 const PlaceOrder: React.FC = () => {
+  const { orderForm } = useOrderForm()
+  const { value, referenceValue, interestValue } = useOrderPayment()
+  const {
+    culture: { currency },
+    rootPath = '',
```
Comment (kevinch) [inline react/PlaceOrder.tsx:17]: Nice.

File: react/PlaceOrder.tsx [hunk 3]
```diff
@@ -1,14 +1,138 @@
...
+      value,
+      referenceValue,
+      interestValue,
+      savePersonalData: true,
+      optinNewsLetter: false,
+    }
+
+    setPlacingOrder(true)
+
+    const startTransactionResponse = await fetch(
```
Comment (kevinch) [inline react/PlaceOrder.tsx:37]: Will you move this to a Graphql call in the future?

File: react/PlaceOrder.tsx [hunk 4]
```diff
@@ -1,14 +1,138 @@
...
+      const callbackResponse = await fetch(
+        `${rootPath}/api/checkout/pub/gatewayCallback/${orderGroupId}`,
+        { method: 'POST' }
+      )
+
+      if (callbackResponse.ok) {
+        window.location.replace(redirectUrl)
+      }
+    } else if (!receiverUri) {
+      // go to first invalid step
```
Comment (kevinch) [inline react/PlaceOrder.tsx:118]: Will this condition receive code late on?

File: react/PlaceOrder.tsx (hunk 1 above)
Comment (lucasecdb) [reply react/PlaceOrder.tsx:22]: it's the typescript interface for the instance of an iframe, so we can get autocomplete on stuff like `cardFormIframe.contentWindow.etc`

File: react/PlaceOrder.tsx (hunk 4 above)
Comment (lucasecdb) [reply react/PlaceOrder.tsx:118]: yes! I just need to figure out what to do in this case 😂

the other sections in this if also needs to be updated, specially for handling error cases

File: react/PlaceOrder.tsx (hunk 3 above)
Comment (lucasecdb) [reply react/PlaceOrder.tsx:37]: no, I don't think it would benefit much here :-/

**Review by jeffersontuc — APPROVED** [review]
It went smoothly 🎉

File: react/PlaceOrder.tsx (hunk 1 above)
Comment (kevinch) [reply react/PlaceOrder.tsx:22]: Nice.

File: react/PlaceOrder.tsx (hunk 4 above)
Comment (kevinch) [reply react/PlaceOrder.tsx:118]: Ok. Do you have an issue/story documenting this?

**Review by kevinch — APPROVED** [review]

File: react/PlaceOrder.tsx (hunk 4 above)
Comment (lucasecdb) [reply react/PlaceOrder.tsx:118]: just created! [CHUI-49](https://vtex-dev.atlassian.net/browse/CHUI-49)

**Pull request merged**